### PR TITLE
Cob/invalidation

### PR DESCRIPTION
**Pull request by coburncoburn**

### CODE REVIEW

File: contracts/AnchoredPriceView.sol [hunk 1]
```diff
@@ -288,6 +295,15 @@ contract AnchoredPriceView {
         revert("Unknown token address");
     }
 
+    function invalidate(bytes memory message, bytes memory signature) public {
+        (string memory decoded_message, address target) = abi.decode(message, (string, address));
+        require(keccak256(abi.encodePacked(decoded_message)) == keccak256(abi.encodePacked("rotate")), "invalid message must be 'rotate'");
+        require(priceData.source(message, signature) == source, "invalidation message must come from the reporter");
+        target; //shh
```
Comment (antonina-cherednichenko) [inline contracts/AnchoredPriceView.sol:302]: curious what this line does?

Comment (coburncoburn) [reply contracts/AnchoredPriceView.sol:302]:
```
➜  open-oracle git:(cob/invalidation) yarn compile
yarn run v1.22.4
warning ../../package.json: No license field
$ npx saddle compile
Compiling contracts contracts/*.sol tests/contracts/*.sol with solc to /Users/coburn/Compound/open-oracle/.build/contracts.json...
Warning: Unused local variable.
   --> contracts/AnchoredPriceView.sol:299:41:
    |
299 |         (string memory decoded_message, address target) = abi.decode(message, (string, address));
    |                                         ^^^^^^^^^^^^^^


```
just to get rid of this unused local variable complaint : /

File: contracts/AnchoredPriceView.sol [hunk 2]
```diff
@@ -221,6 +225,9 @@ contract AnchoredPriceView {
      * @return The price for the given cToken address
      */
     function getUnderlyingPrice(address cToken) external view returns (uint256) {
+        if (breaker == true) {
+            return anchor.getUnderlyingPrice(cToken);
```
Comment (antonina-cherednichenko) [inline contracts/AnchoredPriceView.sol:229]: so, if breaker is on, price in ETH is returned, and if it's off - USD?

Comment (coburncoburn) [reply contracts/AnchoredPriceView.sol:229]: this is true, here, but in anticipation of going for only ETH, right?

File: contracts/AnchoredPriceView.sol (hunk 1 above)
Comment (antonina-cherednichenko) [reply contracts/AnchoredPriceView.sol:302]: wow 🤩

**Comment by coburncoburn** [comment]
@antonina-cherednichenko I don't think this iteration makes sense to be able to turn breaker back off, since the source key writing an invalidation message means , in worst case, they have lost the signing key and a newly configured view would be the way to go in that situation. 

Having said that, the key rotation story is pretty much in the minimal form at this point ( want to rotate keys? get a new view ), so something to keep thinking about for future iterations.

File: contracts/AnchoredPriceView.sol (hunk 1 above)
Comment (jflatow) [reply contracts/AnchoredPriceView.sol:302]: I think on this one you could also do `(string memory decoded_message, ) = ...`

Comment (coburncoburn) [reply contracts/AnchoredPriceView.sol:302]: perfect, I tried (string memory, address ) and that failed so I gave up. Your take works though

**Pull request merged**